token/oauth/oauth2.go

```diff
@@ -136,13 +136,17 @@ func LoadReferrerAndResponseMode(ctx context.Context, db application.DB, state s
 		referrer = ref.Referrer
 		responseMode = ref.ResponseMode
 		err = appl.OauthStates().Delete(ctx, ref.ID)
-		return err
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"state": state,
+				"err":   err,
+			}, "unable to delete oauth state reference")
+			return err
+		}
+
+		return nil
 	})
 	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"state": state,
-			"err":   err,
-		}, "unable to delete oauth state reference")
```

Comment:
@sbose78 
In this transaction either delete or load can fail, but because of this line regardless of what fails, it always prints "unable to delete oauth state reference"

Reply:
That's OK. Because we failed to delete the reference. The reason for this can be an error during loading. The err field will give you details of the reason.

Reply:
Prime purpose of this transaction is to load the reference and not to delete it. We are deleting because once we have used the reference we don't need it anymore.